### PR TITLE
feat(campaign): add campaign endpoints

### PR DESCRIPTION
**Pull request by peterdeme**

### CODE REVIEW

File: campaign.go [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+
+type CampaignData struct {
+	SegmentID         string            `json:"segment_id,omitempty"`
+	SenderID          string            `json:"sender_id,omitempty"`
+	ChannelType       string            `json:"channel_type,omitempty"`
+	Text              string            `json:"text,omitempty"`
+	Defaults          map[string]string `json:"defaults,omitempty"`
+	Filter            SegmentFilter     `json:"filter,omitempty"`
+	Attachments       []*Attachment     `json:"attachments,omitempty"`
+	PushNotifications bool              `json:"push_notifications,omitempty"`
```
Comment (yaziine) [inline campaign.go:36]: PushNotifications will be removed, you can assume this and remove it

Comment (yaziine) [reply campaign.go:36]: actually, we are actively working on campaigns, i suspect changes may come in the future (not 100% sure but there is a risk)), i think better to postpone this PR a bit until we are 100% sure on API contract. @ferhatelmas what do you think?

Comment (ferhatelmas) [reply campaign.go:36]: sg, good to handle untyped languages first

Comment (peterdeme) [reply campaign.go:36]: 👍 those are done.
closing the issue.

**Comment by peterdeme** [comment]
Will be done once finalized

**Pull request closed without merging**